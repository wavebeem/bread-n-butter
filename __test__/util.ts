import * as bnb from "../src/bread-n-butter";

export function snapTest<A>(parser: bnb.Parser<A>, input: string): void {
  expect(parser.parse(input)).toMatchSnapshot(JSON.stringify(input));
}
