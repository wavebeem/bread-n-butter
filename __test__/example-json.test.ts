import JSON from "../examples/json";
import { snapTest } from "./util";

test("json complex", () => {
  snapTest(
    JSON,
    `\
{
  "id": "a thing\\nice\tab",
  "another property!"
    : "also cool"
  , "weird formatting is ok too........ðŸ˜‚": 123.45e1,
  "": [
    true, false, null,
    "",
    " ",
    {},
    {"": {}}
  ]
}
`
  );
});

test("json simple", () => {
  snapTest(JSON, `{"array":[1,"two",null,true,false],"obj":{}}`);
});

test("json multiline", () => {
  snapTest(
    JSON,
    `
{
  "array": [1, "two", null, true, false],
  "obj": {}
}
`
  );
});

test("json multiline", () => {
  snapTest(
    JSON,
    `
{
  "array" : [ 1 , "two", null , true , false ] ,
  "obj" : {


        "key"       : "value"


  }
}
`
  );
});
