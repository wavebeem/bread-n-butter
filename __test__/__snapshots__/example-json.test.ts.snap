// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json complex: "{\\n  \\"id\\": \\"a thing\\\\nice\\tab\\",\\n  \\"another property!\\"\\n    : \\"also cool\\"\\n  , \\"weird formatting is ok too........ðŸ˜‚\\": 123.45e1,\\n  \\"\\": [\\n    true, false, null,\\n    \\"\\",\\n    \\" \\",\\n    {},\\n    {\\"\\": {}}\\n  ]\\n}\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "": Array [
      true,
      false,
      null,
      "",
      " ",
      Object {},
      Object {
        "": Object {},
      },
    ],
    "another property!": "also cool",
    "id": "a thing
ice	ab",
    "weird formatting is ok too........ðŸ˜‚": 1234.5,
  },
}
`;

exports[`json multiline: "\\n{\\n  \\"array\\" : [ 1 , \\"two\\", null , true , false ] ,\\n  \\"obj\\" : {\\n\\n\\n        \\"key\\"       : \\"value\\"\\n\\n\\n  }\\n}\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "array": Array [
      1,
      "two",
      null,
      true,
      false,
    ],
    "obj": Object {
      "key": "value",
    },
  },
}
`;

exports[`json multiline: "\\n{\\n  \\"array\\": [1, \\"two\\", null, true, false],\\n  \\"obj\\": {}\\n}\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "array": Array [
      1,
      "two",
      null,
      true,
      false,
    ],
    "obj": Object {},
  },
}
`;

exports[`json simple: "{\\"array\\":[1,\\"two\\",null,true,false],\\"obj\\":{}}" 1`] = `
Object {
  "type": "ParseOK",
  "value": Object {
    "array": Array [
      1,
      "two",
      null,
      true,
      false,
    ],
    "obj": Object {},
  },
}
`;
